test/e2e/e2e_test.go

```diff
@@ -1,122 +1,285 @@
-/*
-Copyright 2024.
-
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
-
-    http://www.apache.org/licenses/LICENSE-2.0
-
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
-
 package e2e
 
 import (
+	"context"
 	"fmt"
+	"log"
+	"os"
 	"os/exec"
+	"strings"
 	"time"
 
+	"github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
-
 	"github.com/openshift-assisted/cluster-api-agent/test/utils"
+	"github.com/openshift/assisted-service/api/hiveextension/v1beta1"
+	aimodels "github.com/openshift/assisted-service/models"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"libvirt.org/go/libvirt"
+	clusterctlclient "sigs.k8s.io/cluster-api/cmd/clusterctl/client"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/tree"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
 )
 
-const namespace = "cluster-api-agent-system"
+const (
+	defaultNIC = "eno3"
```

Comment:
For future: let's make this configurable or detectable - this interface doesn't exist in the host I'm running on

Reply:
The idea was to automatically detect it, but I think for now we can inject it as env var at least we can run the tests without modifying the code

Reply:
https://github.com/openshift-assisted/cluster-api-agent/pull/58